rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,570 @@
+    private static final ActivationConfig.ForBlock ACTIVATIONS = ActivationConfigsForTest.all().forBlock(0);
+
+    private Repository repository;
+    private BridgeStorageProvider bridgeStorageProvider;
+    private BtcBlockStoreWithCache.Factory btcBlockStoreFactory;
+    private BtcBlockStore btcBlockStore;
+    private BridgeEventLogger bridgeEventLogger;
+    private FeePerKbSupport feePerKbSupport;
+    private Block currentBlock;
+    private StorageAccessor bridgeStorageAccessor;
+    private FederationStorageProvider federationStorageProvider;
+    private FederationSupportImpl federationSupport;
+    private LockingCapSupport lockingCapSupport;
+    private BridgeSupport bridgeSupport;
+
+    @Test
+    void whenAllActivationsArePresentAndFederationChanges_shouldSuccesfullyChangeFederation() throws Exception {
+        // Arrange
+   
+        setUp();
+        // Create a default original federation using the list of UTXOs
+        var originalFederation = createOriginalFederation(ORIGINAL_FEDERATION_MEMBERS);
+        var originalUTXOs = federationStorageProvider.getNewFederationBtcUTXOs(BRIDGE_CONSTANTS.getBtcParams(), ACTIVATIONS);
+       
+        // Act & Assert
+   
+        // Create pending federation using the new federation keys
+        voteToCreateEmptyPendingFederation();
+        voteToAddFederatorPublicKeysToPendingFederation(NEW_FEDERATION_MEMBERS);
+
+        var pendingFederation = federationStorageProvider.getPendingFederation();
+        assertNotNull(pendingFederation);
+        // We extract here because we will need for util methods
+        var newFederation = pendingFederation.buildFederation(
+            Instant.EPOCH, 0L, BRIDGE_CONSTANTS.getFederationConstants(), ACTIVATIONS);
+
+        voteToCommitPendingFederation();
+        // Since Lovell is activated we will commit the proposed federation
+        commitProposedFederation();
+        // Next federation creation block height should be as expected
+        assertLastRetiredFederationP2SHScriptMatchesWithOriginalFederation(
+            originalFederation);
+        // Move blockchain until the activation phase
+        activateNewFederation();
+   
+        assertUTXOsReferenceMovedFromNewToOldFederation(originalUTXOs);
+        assertNewAndOldFederationsHaveExpectedAddress(
+            newFederation.getAddress(), originalFederation.getAddress());
+        assertMigrationHasNotStarted();
+
+        // Move blockchain until the migration phase
+        activateMigration();
+        // Migrate funds
+        migrateUTXOs();
+
+        assertNewAndOldFederationsHaveExpectedAddress(
+            newFederation.getAddress(), originalFederation.getAddress());
+        assertMigrationHasStarted();
+        verifySigHashes();
+        verifyPegoutTransactionCreatedEventWasEmitted();
```

Comment:
There are other events that should be emitted as well

Reply:
I have added the verification for `verifyReleaseBtcRequestedEventEventWasEmitted`. Let me know if that is good. Or you were thinking on something else.

Reply:
release_requested, pegout_confirmed, pegout_transaction_created , addSignature (per signature of the migration transaction), release_btc

I think those are the events emitted for a migration transaction. Check with @jeremy-then if there is any docs that describe the complete process. I might be missing some

Reply:
Updated to include pegout_confirmed event as well. I checked that the addSignature and release_btc events are not being emmitted because the ending migration transaction is never being signed. I also checked on how it was being done [here](https://github.com/rsksmart/rskj/blob/95a8f1ab843e85b46c4842d0d140913422987f2d/rskj-core/src/test/java/co/rsk/peg/federation/PowpegMigrationTest.java#L76) and it is also not adding signatures to the ending migration tx. Should we include this logic here as well?

Reply:
Ok, sounds good. We can look into adding more validations in later stages. Thanks